programmers/20210512/신규-아이디-추천.js

```diff
@@ -0,0 +1,24 @@
+function solution(newId) {
+    //1단계
+    newId = newId.toLowerCase();
+    //2단계
+    newId = newId.match(/[0-9a-z_.-]/g);
+    if (newId === null) return "aaa";
+    newId = newId.join("");
+    //3단계
+    newId = newId.replace(/\.+/g, ".")
+    //4단계
+    newId = newId.replace(/^\.|\.$/g, "");
+    //5단계
+    if (!/[0-9a-z_.-]/.test(newId)) return "aaa";
+    //6단계
+    if (newId.length > 15) {
+        newId = newId.match(/[0-9a-z_.-]{15}/).join();
+        //원래             (/[0-9a-z_.-]{15}/g) 이랬는데 이때 굳이 g를 넣을 필요가 업는건 알겠는데 왜 테스트가 실패했는지는 모르겠습니다.
+        return newId.replace(/\.$/, "");
+    }
+    //7단계
+    const lastString = newId.match(/[0-9a-z_-]$/);
+    while (newId.length < 3) newId += lastString;
+    return newId;
+}
```

Comment:
밑에는 제가 푼 방식은 아닌데 다른 사람이 너무 잘 푼듯하여 참고해서 한번 보시면 좋을거 같아 남깁니다.
```suggestion
}
function solution(new_id) {
    const answer = new_id
        .toLowerCase() // 1
        .replace(/[^\w-_.]/g, '') // 2
        .replace(/\.+/g, '.') // 3
        .replace(/^\.|\.$/g, '') // 4
        .replace(/^$/, 'a') // 5
        .slice(0, 15).replace(/\.$/, ''); // 6
    const len = answer.length;
    return len > 2 ? answer : answer + answer.charAt(len - 1).repeat(3 - len);
}
```